src/Host.cpp

```diff
@@ -2598,3 +2598,13 @@ void Host::setCompactInputLine(const bool state)
         }
     }
 }
+
+QPointer<TConsole> Host::findConsole(QString name)
+{
+    if (name.isEmpty() or name == "main") {
```

Comment:
```suggestion
    if (name.isEmpty() or name == QStringLiteral("main")) {
```

It makes QString() creation 'free' by doing it at compiletime: https://woboq.com/blog/qstringliteral.html

Reply:
I just looked at the generated assembly code. The version with a "naked" C string doesn't actually build a temporary QString each time you use it; instead it generates a call to `QString::compare_helper(QChar const*, int, char const*, int, Qt::CaseSensitivity)`.

That's still a bit more work, and slightly slower code, than simply calling `operator==(QString&,QString&)` but at least it doesn't do the malloc+free dance to create a temporary copy of a 4-character string.

Reply:
I admit that I don't know that much as you do about the low-level details, but https://doc.qt.io/qt-5/qstring.html#QStringLiteral seems pretty convincing to me... :)

Reply:
The link actually seems to recommend `QLatin1String` in this case of = operator use

Reply:
Given that there are a large number (60+) uses of this string "main" in just the `TLuaInterpreter` class IIRC, using `QLatin1String(...)` for this string - that *is* encodable with the Latin1 charset - then the compiler can merge the instances down to a single one - as opposed to having to include 60 instances of it for the `QStringLiteral(...)` case - however there will be a ***small*** (amortised?) run-time construction cost (not sure if it will incurred once or, say, sixty times) compared to a no-runtime-cost for the `QStringLiteral(...)` cases.